code/builders/builder-vite/src/index.ts

```diff
@@ -61,7 +60,7 @@ export const start: ViteBuilder['start'] = async ({
 }) => {
   server = await createViteServer(options as Options, devServer);
 
-  const previewResolvedDir = join(corePath, 'dist/preview');
+  const previewResolvedDir = join((await import('storybook/core-path')).corePath, 'dist/preview');
```

Comment:
This could be reverted if the builder when loaded had it's ESM `dist` loaded instead of it's `CJS`.

This is to ensure `'storybook/core-path'` is resolved to ESM.